modules/sc-mesh-secure-deployment/src/nats/tests/test_cbma_adaptation.py

```diff
@@ -0,0 +1,611 @@
+        cls.mock_comms_ctrl_instance = cls.patcher_comms_ctrl.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        # Delete created yaml/crt files
+        for file_name in [
+            "ms_config.yaml",
+            "ms_config2.yaml",
+            "invalid_yaml_file.yaml",
+            "ms_config4.yaml",
+            "ms_config5.yaml",
+            "ms_config6.yaml",
+            "./MAC/fake_mac.crt",
+        ]:
+            try:
+                os.remove(file_name)
+            except FileNotFoundError:
+                print(f"File {file_name} not found, skipping deletion.")
+            except Exception as e:
+                print(f"Error deleting file {file_name}: {e}")
+
+        # Delete folder
+        try:
+            os.rmdir(cls.folder)
+        except Exception as e:
+            print(f"Error deleting folder {cls.folder}: {e}")
+
+        # Delete interfaces using settings from the list
+        for settings in cls.INTERFACE_SETTINGS:
+            try:
+                ifname = settings.get("ifname")
+                subprocess.run(["ip", "link", "delete", ifname], check=True)
+            except subprocess.CalledProcessError as e:
+                print(f"Error deleting interface {ifname}: {e}")
+            except Exception as e:
+                print(f"Unexpected error deleting interface {ifname}: {e}")
+
+        # Stop patches
+        cls.patcher_has_certificate.stop()
+        cls.patcher_add_interface.stop()
+        cls.patcher_get_base_mtu_size.stop()
+        cls.patcher_comms_ctrl.stop()
+
+        # Clean up logging resources
+        cls.logger.removeHandler(cls.handler)
+        cls.handler.close()
+
+    def setUp(self):
+        # Instantiates CBMAAdaptation with ms_config.yaml. Used for most of the tests.
+        # Tip: comment following line to get traces from CBMA adaptation. Note that it
+        # will break some of the tests that relies on logger outputs.
+        self.logger = MagicMock()
+
+        self.cbma_adaptation = CBMAAdaptation(
+            self.mock_comms_ctrl, self.logger, self.mock_lock, "ms_config.yaml"
+        )
+        # Allow "dummy" interfaces during unit testing
+        self.cbma_adaptation.ALLOWED_KIND_LIST.add("dummy")
+
+    def tearDown(self):
+        self.cbma_adaptation.stop_cbma()
+
+    def test_setup_cbma(self):
+        # Patch CBMAAdaptation.__get_mac_addr() to return ifdummy0_black
+        # interface's MAC that will be given as a parameter to create MAC
+        # for lower batman
+        with patch.object(
+            CBMAAdaptation,
+            "_CBMAAdaptation__get_mac_addr",
+            return_value="12:34:56:78:90:ab",
+        ):
+            result = self.cbma_adaptation.setup_cbma()
+            self.assertTrue(result)
+
+    def test_stop_cbma(self):
+        result = self.cbma_adaptation.stop_cbma()
+        self.assertTrue(result)
+
+    def test_setup__and_stop_cbma2(self):
```

Comment:
There's as double `_` in `test_setup__and_stop_cbma2` :grin:

Reply:
Good eyes! Fixed.